corehq/apps/dump_reload/sql/load.py

```diff
@@ -129,16 +129,25 @@ def load_data_for_db(db_alias, objects):
     with connection.constraint_checks_disabled():
         for obj in PythonDeserializer(objects, using=db_alias):
             if router.allow_migrate_model(db_alias, obj.object.__class__):
+                app = obj.object._meta.app_label,
+                name = obj.object._meta.object_name,
+                pk = obj.object.pk,
                 model_counter.update([obj.object.__class__])
+
+                # DomainOrNullFilter will mean that the dump can
+                # include existing records. We'll need to skip those.
+                Model = type(obj.object)
+                queryset = Model.objects.using(db_alias)
+                if queryset.filter(pk=obj.object.pk).exists():
```

Comment:
Is this check going to noticeably slow down the import process?  Would it be valid to instead catch this in the `try/except` block around `obj.save`?

Reply:
The `try/except` block around `obj.save` only gets triggered because the `with connection.constraint_checks_disabled()` on line 129 doesn't work (for Postgres; it works for MySQL). I've fixed this locally (still need to push those commits to #27862 ), but we can't actually catch the error until the whole DB/shard is loaded.

But! If we don't dump instances that might already exist, I'll drop this.